src/utils/did.rs

```diff
@@ -10,6 +10,11 @@ use crate::{
     utils::crypto::{AsKnownKeyPair, KnownKeyAlg, KnownKeyPair},
 };
 
+pub(crate) fn is_did(did: &str) -> bool {
+    let parts: Vec<String> = did.split(':').map(str::to_string).collect();
```

Comment:
`map(str::to_string)`

I don't think we need strings re-allocation on such simple validation. You can directly compare string slices.

Reply:
```let parts: Vec<_> = did.split(':').collect();```